packages/solidity-contracts/deploy/beta5/005.fuel_message_portal_v2.ts

```diff
@@ -0,0 +1,40 @@
+import type { HardhatRuntimeEnvironment } from 'hardhat/types';
+import type { DeployFunction } from 'hardhat-deploy/dist/types';
+
+import { FuelMessagePortalV2__factory as FuelMessagePortalV2 } from '../../typechain';
+
+const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
+  const {
+    ethers,
+    upgrades: { upgradeProxy, erc1967 },
+    deployments: { get, save },
+  } = hre;
+  const [deployer] = await ethers.getSigners();
+
+  const fuelMessagePortal = await get('FuelMessagePortal');
+
+  const { deployTransaction, address } = await upgradeProxy(
+    fuelMessagePortal,
+    new FuelMessagePortalV2(deployer),
+    {
+      unsafeAllow: ['constructor'],
```

Comment:
What is this unsafe?

Reply:
The use of constructors in upgradable contracts. It is disabled by default, but if you want to be able to use `immutables` in the upgradable contracts, this is the way.